toolchain/check/subst.h

```diff
@@ -20,6 +21,24 @@ struct Substitution {
 
 using Substitutions = llvm::ArrayRef<Substitution>;
 
+// A function that performs any needed substitution into an instruction, as part
+// of substituting into a symbolic instruction sequence. The instruction ID
+// should be updated as necessary to represent the new instruction. If the
+// function returns true, the instruction is treated as fully-substituted; if it
+// returns false, the instruction will be decomposed and substitution will be
+// performed recursively into its operands.
```

Comment:
Most of the comments seem to be describing implementation constraints. From reading the implementation, I think this is different: is this a constraint on the caller, or is it a constraint on the implementation? If it's a constraint on the caller, perhaps it should be on SubstInst instead?

Reply:
Changed to talk more about what the return value means and less about what `SubstInst` will do with it.